model/src/cfd.rs

```diff
@@ -3100,6 +3140,86 @@ mod tests {
         assert_eq!(is_liquidation_price, should_liquidation_price);
     }
 
+    #[test]
+    fn given_settlement_in_24_hours_when_calculating_hours_to_extend_based_on_event_then_return_expected_hours(
+    ) {
+        let settlement_interval = SETTLEMENT_INTERVAL.whole_hours();
+
+        let candidate_event_id = BitMexPriceEventId::with_20_digits(
+            OffsetDateTime::now_utc() + settlement_interval.hours(),
+        );
+
+        for hour in 0..settlement_interval {
+            let current_event_id =
+                BitMexPriceEventId::with_20_digits(OffsetDateTime::now_utc() + hour.hours());
+
+            let taker = Cfd::dummy_taker_long().dummy_open(current_event_id);
+            let maker = Cfd::dummy_maker_short().dummy_open(current_event_id);
+
+            assert_eq!(
+                taker
+                    .hours_to_extend_in_rollover_based_on_event(candidate_event_id)
+                    .unwrap(),
+                (candidate_event_id.timestamp() - current_event_id.timestamp()).whole_hours()
+                    as u64
+            );
+
+            assert_eq!(
+                maker
+                    .hours_to_extend_in_rollover_based_on_event(candidate_event_id)
+                    .unwrap(),
+                (candidate_event_id.timestamp() - current_event_id.timestamp()).whole_hours()
+                    as u64
+            );
+        }
```

Comment:
I find it hard to reason about asserts within a for loop. What do you think of adding one additional positive test case for a value between `0` and `settlement_interval`? It might be redundant but can help readability.

Reply:
Are you suggesting to add a separate test function which is the same as this one but for only a single case? I can do that.

Reply:
yes please.